src/main/kotlin/com/exactpro/th2/common/schema/message/impl/rabbitmq/configuration/RabbitMQConfiguration.kt

```diff
@@ -18,31 +18,84 @@ package com.exactpro.th2.common.schema.message.impl.rabbitmq.configuration
-    var subscriberName: String? = null,  //FIXME: Remove in future version
-    var exchangeName: String? = null) : Configuration()
+    val subscriberName: String? = null,  //FIXME: Remove in future version
+    val exchangeName: String? = null
+) : Configuration()
 
 data class ConnectionManagerConfiguration(
-    var subscriberName: String? = null,
-    var connectionTimeout: Int = -1,
-    var connectionCloseTimeout: Int = 10000,
-    var maxRecoveryAttempts: Int = 5,
-    var minConnectionRecoveryTimeout: Int = 10000,
-    var maxConnectionRecoveryTimeout: Int = 60000,
+    val subscriberName: String? = null,
+    val connectionTimeout: Int = -1,
+    val connectionCloseTimeout: Int = 10000,
+    val maxRecoveryAttempts: Int = 5,
+    val minConnectionRecoveryTimeout: Int = 10000,
+    val maxConnectionRecoveryTimeout: Int = 60000,
     val prefetchCount: Int = 10,
+    val retryTimeDeviationPercent: Int = 10,
     val messageRecursionLimit: Int = 100,
     val workingThreads: Int = 1,
     val confirmationTimeout: Duration = Duration.ofMinutes(5)
 ) : Configuration() {
     init {
+        check(maxRecoveryAttempts > 0) { "expected 'maxRecoveryAttempts' greater than 0 but was $maxRecoveryAttempts" }
+        check(minConnectionRecoveryTimeout > 0) { "expected 'minConnectionRecoveryTimeout' greater than 0 but was $minConnectionRecoveryTimeout" }
+        check(maxConnectionRecoveryTimeout >= minConnectionRecoveryTimeout) { "expected 'maxConnectionRecoveryTimeout' ($maxConnectionRecoveryTimeout) no less than 'minConnectionRecoveryTimeout' ($minConnectionRecoveryTimeout)" }
         check(workingThreads > 0) { "expected 'workingThreads' greater than 0 but was $workingThreads" }
         check(!confirmationTimeout.run { isNegative || isZero }) { "expected 'confirmationTimeout' greater than 0 but was $confirmationTimeout" }
     }
+
+    fun createRetryingDelaySequence(): Sequence<RetryingDelay> {
+        return generateSequence(RetryingDelay(0, minConnectionRecoveryTimeout)) {
+            RetryingDelay(it.tryNumber + 1, RetryingDelay.getRecoveryDelay(
+                it.tryNumber + 1,
+                minConnectionRecoveryTimeout,
+                maxConnectionRecoveryTimeout,
+                maxRecoveryAttempts,
+                retryTimeDeviationPercent
+            ))
+        }
+    }
+}
+
+data class RetryingDelay(val tryNumber: Int, val delay: Int) {
+    companion object {
+        @JvmStatic
+        fun getRecoveryDelay(
+            numberOfTries: Int,
+            minTime: Int,
+            maxTime: Int,
+            maxRecoveryAttempts: Int,
+            deviationPercent: Int
+        ): Int {
+            return if (numberOfTries <= maxRecoveryAttempts) {
+                getRecoveryDelayWithIncrement(numberOfTries, minTime, maxTime, maxRecoveryAttempts)
+            } else {
+                getRecoveryDelayWithDeviation(maxTime, deviationPercent)
+            }
```

Comment:
Maybe use divination in all cases because when several components lost connection at the same time, after recovery RabbitMQ will receive several reconnect requests at the same time.
It is not critical problem, but we should keep in mind